src/plugins/irc-events/link.js

```diff
@@ -207,11 +207,22 @@ function disablePreviewDueBlacklist(msg, chan, network) {
 	const sender = msg.from.nick;
 	const text = msg.text;
 
+	function testAgainstRegex(query, string) {
+		try {
+			const match = /^\/(.*)\/([a-z]*)$/.exec(query);
+			return new RegExp(match[1], match[2]).test(string);
+		} catch (e) {
+			// Invalid regex expression
+			return false;
+		}
+	}
+
 	for (const query of network.mediaPreviewBlacklist) {
 		if (
 			(query[0] === "#" && query === channelName) ||
 			(query[0] === "@" && query.slice(1) === sender) ||
-			(!["@", "#"].includes(query[0]) && text.indexOf(query) !== -1)
+			(!["@", "#"].includes(query[0]) && text.indexOf(query) !== -1) ||
+			(query[0] === "/" && testAgainstRegex(query, text))
 		) {
```

Comment:
Can we restructure this so that we parse the blacklist exactly once?
As is, we parse the regex on every single message that contains a url.
Also there's no way to tell the user that the regex the entered is wrong (we are simply ignoring the error here).

In my view it would be better to parse this exactly once, when the settings are entered. This allows us to give an error message back to the user for anything that starts with the regex trigger but doesn't parse.